packages/cli/docs-preview/src/runAppPreviewServer.ts

```diff
@@ -168,18 +168,60 @@ export async function runAppPreviewServer({
         if (!serverProcess.killed) {
             context.logger.debug(`Killing server process with PID: ${serverProcess.pid}`);
             try {
+                // First try graceful shutdown
                 serverProcess.kill();
+
+                // If process doesn't exit within 2 seconds, force kill
+                setTimeout(() => {
+                    if (!serverProcess.killed) {
+                        context.logger.debug(`Force killing server process with PID: ${serverProcess.pid}`);
+                        try {
+                            serverProcess.kill("SIGKILL");
+                        } catch (err) {
+                            context.logger.error(`Failed to force kill server process: ${err}`);
+                        }
+                    }
+                }, 2000);
             } catch (err) {
                 context.logger.error(`Failed to kill server process: ${err}`);
             }
         }
     };
 
-    // clean up process
-    process.on("SIGINT", cleanup);
-    process.on("SIGTERM", cleanup);
+    // handle termination signals
+    const shutdownSignals = ["SIGTERM", "SIGINT"];
+    const failureSignals = ["SIGHUP"];
+    for (const shutSig of shutdownSignals) {
+        process.on(shutSig, () => {
+            context.logger.debug("Shutting down server...");
+            cleanup();
+        });
+    }
+    for (const failSig of failureSignals) {
+        process.on(failSig, () => {
+            context.logger.error("Server failed, shutting down process...");
+            cleanup();
+        });
+    }
+
+    // handle normal exit
     process.on("exit", cleanup);
 
+    // handle uncaught exits
+    process.on("uncaughtException", (err) => {
+        context.logger.error(`Uncaught exception: ${err}`);
```

Comment:
what does the message here end up looking like? also not that useful to a user or `${err}` could be pretty ugly